### ISSUE TITLE
additionalField(s) missing cache

### ISSUE THREAD

**Issue by mrblur** [nao-pon/flysystem-google-drive#31]
Hi,

Just enabled caching with Symfony's `cache.app` adapter and noticed that `getContents` listings read from cache are missing additional fields.
I am fetching `hasThumbnail` with `thumbnailLink` fine from API, but miss them when read back from cache.

Is this a Flysystem's limitation?

**Comment by nao-pon** [comment]
@mrblur Yes, it is probably a limitation of the FlySystem's cache mechanism. You may need to extend the cache mechanism.

**Comment by nao-pon** [comment]
There is extra cache trait for param `hasDir`. https://github.com/nao-pon/flysystem-cached-extra/blob/master/src/Hasdir.php
